omni/omni.py

```
@@ -0,0 +1,967 @@
+                        data = await response.json()
+                        result = data.get("results", [{}])[0]
+                        category_scores = result.get("category_scores", {})
+                        return category_scores
+                    elif response.status == 500:
+                        await asyncio.sleep(5)  # Wait a few seconds before retrying
+                    else:
+                        # Log the error if the request failed
+                        await self.log_message(message, {}, error_code=response.status)
+                        return {}
+        except Exception as e:
+            raise RuntimeError(f"Failed to analyze content: {e}")
```

Ruff (0.8.2)

Within an `except` clause, raise exceptions with `raise ... from err` or `raise ... from None` to distinguish them from errors in exception handling

(B904)